crates/core/src/nats/client_opts.rs

```diff
@@ -0,0 +1,93 @@
+use std::time::Duration;
+
+use async_nats::ConnectOptions;
+
+use super::{types::AsyncNatsClient, ConnId, NatsError};
+
+#[derive(Debug, Clone, Default)]
+pub enum NatsUserRole {
+    Admin,
+    #[default]
+    Public,
+}
+
+#[derive(Debug, Clone)]
+pub struct ClientOpts {
+    pub(crate) url: String,
+    pub(crate) role: NatsUserRole,
+    // This ID is being used as name within the ConnectOptions
+    pub(crate) conn_id: ConnId,
+    // This is being used as prefix for nats streams, consumers and subjects names
+    pub(crate) nats_prefix: String,
+    pub(crate) timeout_secs: u64,
+}
+
+impl ClientOpts {
+    pub fn new(url: impl ToString, conn_id: ConnId) -> Self {
+        Self {
+            url: url.to_string(),
+            nats_prefix: conn_id.clone().into(),
+            conn_id,
+            role: NatsUserRole::default(),
+            timeout_secs: 5,
+        }
+    }
+
+    pub fn public_opts(url: impl ToString, conn_id: ConnId) -> Self {
+        Self::new(url, conn_id).with_role(NatsUserRole::Public)
+    }
+
+    pub fn admin_opts(url: impl ToString, conn_id: ConnId) -> Self {
```

Comment:
Should these accept a `conn_id` arg ? End users should be able to compose a `ClientOpts` without worrying about the `conn_id`. It's also a security leak. Let's encapsulate the `conn_id` in `ClientOpts` to avoid this completely.